src/provider/rpc.ts

```diff
@@ -34,8 +35,22 @@ export class RpcProvider implements ProviderInterface {
 
   public channel: RpcChannel;
 
-  constructor(optionsOrProvider?: RpcProviderOptions | ProviderInterface | RpcProvider) {
-    if (optionsOrProvider && 'channel' in optionsOrProvider) {
+  constructor(
+    optionsOrProvider?: RpcProviderOptions | ProviderInterface | RpcProvider,
+    instance?: boolean
+  ) {
+    if (optionsOrProvider && instance) {
+      // stop ts complains
+      // @ts-expect-error
+      this.channel = null;
```

Comment:
Could also be done with a cast.

```suggestion
      this.channel = null as any;
```